examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md

```diff
@@ -0,0 +1,359 @@
+
+## Setting up OAuth Scopes
+
+Next, configure the scopes (or services) that the OAuth client ID will have access to. In this case, we’ll configure access to the Google Calendar API.
+
+- In the left menu click **Data Access**
+- Click **Add or Remove Scopes**
+- In the right panel filter on `https://www.googleapis.com/auth/calendar`
+- In the filtered results, choose the first result, the scope should end with `/auth/calendar`
+- Click **Update** and then **Save**
+
+
+<!--ARCADE EMBED START--><div style="position: relative; padding-bottom: calc(53.793774319066145% + 41px); height: 0; width: 100%;"><iframe src="https://demo.arcade.software/mbsRtOs10arPZtzjeum2?embed&embed_mobile=tab&embed_desktop=inline&show_copy_link=true" title="Clients – Google Auth Platform – cookbook-demo – Google Cloud console" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen allow="clipboard-write" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; color-scheme: light;" ></iframe></div><!--ARCADE EMBED END-->
+
+# GPT Action Configuration Steps
+
+We are now ready to configure the GPT Action. First we'll configure the OAuth settings to allow the GPT to authenticate with Google Calendar. 
+
+- In your GPT, create an action. 
+- Click on the settings gear icon and select **OAuth**
+- Enter the **Client ID** and **Client Secret** from the Google Cloud Console. 
+- Enter the following details: 
+  - Authorization URL: `https://accounts.google.com/o/oauth2/auth`
+  - Token URL: `https://oauth2.googleapis.com/token`
+  - Scopes: `https://www.googleapis.com/auth/calendar`
+- Leave the Token Exchange Method as default. 
+- Click **Save**
+
+
+<img src="../../../images/google-calendar-action-config.png" alt="Google Calendar OAuth" width="400"/>
+
+We can now enter the OpenAPI schema for the action. The config below allows reading and creating events. Enter this in the OpenAPI schema field.
+
+```yaml
+openapi: 3.1.0
+info:
+  title: Google Calendar API
+  description: This API allows you to read and create events in a user's Google Calendar.
+  version: 1.0.0
+  contact:
+    name: Your Name or Company
+    email: your.email@example.com
+    url: https://www.yourcompany.com
+servers:
+  - url: https://www.googleapis.com/calendar/v3
+    description: Google Calendar API server
+
+paths:
+  /calendars/primary/events:
+    get:
+      summary: List events from the primary calendar
+      description: Retrieve a list of events from the user's primary Google Calendar.
+      operationId: listEvents
+      tags:
+        - Calendar
+      security:
```

Comment:
I believe security isn't needed as its handled by the GPT

Reply:
Good spot, removed.